JPA sample can't create tables

`Caused by: java.sql.SQLException: Storage engine MyISAM is disabled (Table creation is disallowed).`

It appears that something maybe in SB2 changed JPAs dialect to something that Cloud SQL doesn't like.

https://cloud.google.com/sql/docs/mysql/diagnose-issues (search for MyISAM).

I've tried a few dialects, but haven't found one that works yet.

`spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL5Dialect`
`spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQLInnoDBDialect`

It should be a matter of finding one that Cloud SQL likes.
https://docs.jboss.org/hibernate/orm/3.5/api/org/hibernate/dialect/package-summary.html

--- a/spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/src/main/resources/application.properties
+++ b/spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/src/main/resources/application.properties
@@ -4,6 +4,6 @@ spring.cloud.gcp.sql.database-name=
 spring.jpa.hibernate.ddl-auto=create-drop
 
 # Cloud SQL (MySQL) only supports InnoDB, not MyISAM
-spring.jpa.database-platform=org.hibernate.dialect.MySQL57Dialect
+spring.jpa.database-platform=org.hibernate.dialect.MySQL55Dialect
 
 #spring.cloud.gcp.credentials.location=file:/

--- a/spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/src/main/resources/application.properties
+++ b/spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/src/main/resources/application.properties
@@ -3,4 +3,7 @@ spring.cloud.gcp.sql.database-name=
 
 spring.jpa.hibernate.ddl-auto=create-drop
 
+# Cloud SQL (MySQL) only supports InnoDB, not MyISAM
+spring.jpa.database-platform=org.hibernate.dialect.MySQL5InnoDBDialect
+
 #spring.cloud.gcp.credentials.location=file:/

--- a/spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/src/main/resources/application.properties
+++ b/spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/src/main/resources/application.properties
@@ -4,6 +4,6 @@ spring.cloud.gcp.sql.database-name=
 spring.jpa.hibernate.ddl-auto=create-drop
 
 # Cloud SQL (MySQL) only supports InnoDB, not MyISAM
-spring.jpa.database-platform=org.hibernate.dialect.MySQL5InnoDBDialect
+spring.jpa.database-platform=org.hibernate.dialect.MySQL57Dialect
 
 #spring.cloud.gcp.credentials.location=file:/

